distributed/tests/test_nanny.py

```diff
@@ -290,6 +290,58 @@ def leak():
         assert "memory" in out.lower()
 
 
+@gen_cluster(
+    nthreads=[("127.0.0.1", 1)] * 8,
+    client=True,
+    Worker=Nanny,
+    worker_kwargs={"memory_limit": 2e8, "memory_pause_fraction": 0.6},
+    timeout=20,
+    clean_kwargs={"threads": False},
+)
+def test_nanny_throttle(c, s, *workers):
+    def data(size, i):
+        return b"0" * size
+
+    a = workers[0]
+    proc = a.process.pid
+    from distributed import wait
+
+    with captured_logger(logging.getLogger("distributed.worker")) as logger:
+        size = int(0.05 * a.memory_limit)
+        futures = [
+            c.submit(data, size, i, workers=[a.worker_address]) for i in range(12)
+        ]
+        yield wait(futures)
+
+        def check(dask_worker):
+            self = dask_worker
+            proc = self.monitor.proc
+            memory = proc.memory_info().rss
+            frac = memory / self.memory_limit
+            return frac, self.paused, self.memory_pause_fraction
+
+        start = time()
```

Comment:
Is there a helper for this type of thing? "Keep repeating this check until it passes, else raise when you hit the timeout?"

Reply:
I think that there is, but I tend to just use a `while True` loop because I understand it more.